### PR TITLE
Added the serviceAccount authentication to BigQuery

### PR DESCRIPTION
**Pull request by hariss63**

### CODE REVIEW

File: src/main/java/org/wso2/carbon/connector/BigQueryConstant.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+public class BigQueryConstant {
+
+    public static final String keyAlias = "privatekey";
+    public static final String signature = "SHA256withRSA";
+    public static final String keyStore = "PKCS12";
+    public static final String keyStoreLocation = "keyStoreLocation";
+    public static final String serviceAccount = "serviceAccount";
+    public static final String scope = "scope";
+    public static final String password = "notasecret";
```
Comment (RKathees) [inline src/main/java/org/wso2/carbon/connector/BigQueryConstant.java]: Constant variables should be in capital.

File: src/main/java/org/wso2/carbon/connector/CreateJWT.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+ */
+public class CreateJWT extends AbstractConnector {
+    private static final Log log = LogFactory.getLog(CreateJWT.class);
+    public static String keyAlias = JWTConstant.KEYALIAS;
+
+    public void connect(MessageContext messageContext) {
+        try {
+            String token = getJsonWebToken(messageContext);
+            messageContext.setProperty(JWTConstant.JWT_PROP, token);
+
```
Comment (malakasilva) [inline src/main/java/org/wso2/carbon/connector/CreateJWT.java:55]: Extra spacing

Comment (hariss63) [inline src/main/java/org/wso2/carbon/connector/CreateJWT.java:55]: Fixed in https://github.com/wso2-extensions/esb-connector-bigquery/pull/11

File: src/main/java/org/wso2/carbon/connector/CreateJWT.java [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        }
+
+        byte[] encodedHeader = new byte[0];
+        if (jwtHeaderStr != null) {
+            encodedHeader = Base64.encodeBase64(jwtHeaderStr.getBytes(JWTConstant.UTF));
+        }
+
+        //Construct JWT Claim
+        JSONObject jwtClaimSet = new JSONObject();
+        long iat = (System.currentTimeMillis() / 1000) - 60;
```
Comment (dinusha-dilrukshi) [inline src/main/java/org/wso2/carbon/connector/CreateJWT.java:147]: Use descriptive variable names.

File: src/main/java/org/wso2/carbon/connector/CreateJWT.java [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        long exp = iat + 3600;
```
Comment (dinusha-dilrukshi) [inline src/main/java/org/wso2/carbon/connector/CreateJWT.java:148]: Better take 3600 into a constant, if expiry is not configurable.

File: src/main/java/org/wso2/carbon/connector/CreateJWT.java (hunk 3 above)
Comment (hariss63) [reply src/main/java/org/wso2/carbon/connector/CreateJWT.java:147]: Fixed in https://github.com/wso2-extensions/esb-connector-bigquery/pull/11/commits/074492ecf2bf9deab34a477ce509b792eb891ca3

File: src/main/java/org/wso2/carbon/connector/CreateJWT.java (hunk 4 above)
Comment (hariss63) [reply src/main/java/org/wso2/carbon/connector/CreateJWT.java:148]: Fixed in https://github.com/wso2-extensions/esb-connector-bigquery/pull/11/commits/074492ecf2bf9deab34a477ce509b792eb891ca3

File: src/main/java/org/wso2/carbon/connector/CreateJWT.java [hunk 5]
```diff
@@ -0,0 +1,182 @@
+package org.wso2.carbon.connector;
+
```
Comment (Jasintha) [inline src/main/java/org/wso2/carbon/connector/CreateJWT.java:2]: copyright license is usually put at the beginning of every source file

**Comment by hariss63** [comment]
Fixed in latest connector repo https://github.com/wso2-extensions/esb-connector-bigquery/blob/master/src/main/java/org/wso2/carbon/connector/CreateJWT.java

**Pull request merged**